packages/core/src/network.js

```diff
@@ -194,6 +194,21 @@ export class Network {
     // do not handle data urls
     if (request.url.startsWith('data:')) return;
 
+    // Network requests are handled by the browser, which is lenient about URL encoding.
+    // Invalid characters in the URL won't cause snapshot/build failures.
+    // Therefore, this code only checks for invalid characters in the network request URL.
+    try {
+      let modifiedURL = modifyURL(request.url);
+      if (modifiedURL !== request.url) {
+        this.log.debug(`Inconsistent network URL detected for url: ${request.url}.`);
```

Comment:
Check if this will fail on jackproxy - if not then dont show error - if yes then show. 

And also move this to utils and pass logger

Reply:
- Tried with different invalid characters eg. `?? @ !`. But for some of the special characters `%`, `%11 `, or leading spaces at the end of URL. ie. `http//localhost:8000/test% .png`, for these cases it's passed in the browser but our jackproxy breaks. These cases can be detected by `decodeAndEncodeURLWithLogging` functions.

Reply:
Here, I test with this url:` http://localhost:3000/avif_edge_support/testfile%_5232.png`, that load fine in browser, but causes error on render side. `Proxy fails to start!`.